src/services/db/GitFileSystemService.ts

```diff
@@ -286,33 +338,107 @@ export default class GitFileSystemService {
 
   // Clone repository from upstream Git hosting provider
   clone(repoName: string): ResultAsync<string, GitFileSystemError> {
-    const originUrl = `git@github.com:${ISOMER_GITHUB_ORG_NAME}/${repoName}.git`
+    return combine([
+      this.cloneStaging(repoName),
+      this.cloneStagingLite(repoName),
+    ]).andThen(([stagingPath, _]) =>
+      // staging lite path not needed, promises are resolved in order
+      okAsync(stagingPath)
+    )
+  }
 
-    return this.getFilePathStats(repoName, "")
+  cloneStaging(repoName: string): ResultAsync<string, GitFileSystemError> {
+    const originUrl = `git@github.com:${ISOMER_GITHUB_ORG_NAME}/${repoName}.git`
+    return this.getFilePathStats(repoName, "", true)
       .andThen((stats) => ok(stats.isDirectory()))
       .orElse((error) => {
         if (error instanceof NotFoundError) {
           return ok(false)
         }
         return err(error)
       })
+
       .andThen((isDirectory) => {
         if (!isDirectory) {
           return ResultAsync.fromPromise(
             this.git
-              .clone(originUrl, `${EFS_VOL_PATH}/${repoName}`)
-              .cwd(`${EFS_VOL_PATH}/${repoName}`)
-              .checkout(BRANCH_REF),
+              .clone(originUrl, `${EFS_VOL_PATH_STAGING}/${repoName}`)
+              .cwd(`${EFS_VOL_PATH_STAGING}/${repoName}`)
+              .checkout(STAGING_BRANCH),
             (error) => {
               logger.error(`Error when cloning ${repoName}: ${error}`)
 
               if (error instanceof GitError) {
-                return new GitFileSystemError("Unable to clone repo")
+                return new GitFileSystemError("Unable to clone whole repo")
+              }
+
+              return new GitFileSystemError("An unknown error occurred")
+            }
+          ).map(() => `${EFS_VOL_PATH_STAGING}/${repoName}`)
+        }
+
+        return this.isGitInitialized(repoName)
+          .andThen((isGitInitialized) => {
+            if (!isGitInitialized) {
+              return errAsync(
+                new GitFileSystemError(
+                  `An existing folder "${repoName}" exists in staging but is not a Git repo`
+                )
+              )
+            }
+            return okAsync(true)
+          })
+          .andThen(() => this.isOriginRemoteCorrect(repoName))
+          .andThen((isOriginRemoteCorrect) => {
+            if (!isOriginRemoteCorrect) {
+              return errAsync(
+                new GitFileSystemError(
+                  `An existing folder "${repoName}" exists in staging lite but is not the correct Git repo`
+                )
+              )
+            }
+            return okAsync(`${EFS_VOL_PATH_STAGING}/${repoName}`)
+          })
+      })
+  }
+
+  cloneStagingLite(repoName: string): ResultAsync<string, GitFileSystemError> {
```

Comment:
Do we really need 2 separate methods to clone for lite vs staging? seems like 80% logic will be the same right?

Reply:
derped on this, yup, better to combine b307b4b493b06e45da22d4137b8994865b9e99ed